### PR TITLE
PSA: update EC curve and DH group family macros

### PR DESCRIPTION
**Pull request by paul-elliott-arm**
## Description

Fixes [Issue #3273 ](https://github.com/ARMmbed/mbedtls/issues/3273)

To resolve descrepancy between macro names in Mbed TLS and PSA Crypto API 1.0.0:

- Rename `PSA_ECC_CURVE_xxx` to `PSA_ECC_FAMILY_xxx`
- Rename `PSA_KEY_TYPE_GET_CURVE` to `PSA_KEY_TYPE_ECC_GET_FAMILY`
- Rename `psa_ecc_curve_t` to `psa_ecc_family_t`

- Rename `PSA_DH_GROUP_xxx` to  `PSA_DH_FAMILY_xxx`
- Rename `PSA_KEY_TYPE_GET_GROUP` to `PSA_KEY_TYPE_DH_GET_FAMILY`
- Rename `psa_dh_group_t` to `psa_dh_family_t`

All old names are provided as deprecated in crypto_compat.h for backward compatibility.

## Status
READY

## Requires Backporting
NO  

## Migrations
YES 

Although this is an API change, it is one to make sure we adhere to the PSA spec, and thus necessary.

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
pr-head failed due to a network glitch in one job, but since that job passed in pr-merge, that's ok.

pr-merge only failed because of an ABI change in an internal function, so that's a pass.

File: ChangeLog.d/psa_ecc_macros.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+API changes
+   * Rename PSA_ECC_CURVE_xxx to PSA_ECC_FAMILY_xxx in line with PSA spec
+   * Rename PSA_KEY_GET_CURVE to to PSA_KEY_TYPE_ECC_GET_FAMILY and rename
```
Comment (gilles-peskine-arm) [inline ChangeLog.d/psa_ecc_macros.txt:3]:
```suggestion
   * Rename PSA_KEY_TYPE_GET_CURVE to to PSA_KEY_TYPE_ECC_GET_FAMILY and rename
```
and likewise for DH

File: ChangeLog.d/psa_ecc_macros.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+API changes
+   * Rename PSA_ECC_CURVE_xxx to PSA_ECC_FAMILY_xxx in line with PSA spec
```
Comment (gilles-peskine-arm) [inline ChangeLog.d/psa_ecc_macros.txt:2]: Please don't use unnecessary abbreviations like “spec” for “specification”.

Comment (gilles-peskine-arm) [inline ChangeLog.d/psa_ecc_macros.txt:2]: Please put a final stop at the end of the sentence.

File: include/psa/crypto_compat.h [hunk 3]
```diff
@@ -50,8 +50,8 @@ extern "C" {
 typedef MBEDTLS_PSA_DEPRECATED size_t mbedtls_deprecated_size_t;
 typedef MBEDTLS_PSA_DEPRECATED psa_status_t mbedtls_deprecated_psa_status_t;
 typedef MBEDTLS_PSA_DEPRECATED psa_key_usage_t mbedtls_deprecated_psa_key_usage_t;
-typedef MBEDTLS_PSA_DEPRECATED psa_ecc_curve_t mbedtls_deprecated_psa_ecc_curve_t;
-typedef MBEDTLS_PSA_DEPRECATED psa_dh_group_t mbedtls_deprecated_psa_dh_group_t;
+typedef MBEDTLS_PSA_DEPRECATED psa_ecc_family_t mbedtls_deprecated_psa_ecc_family_t;
+typedef MBEDTLS_PSA_DEPRECATED psa_dh_family_t mbedtls_deprecated_psa_dh_family_t;
```
Comment (gilles-peskine-arm) [inline include/psa/crypto_compat.h:54]: Missing: definitions of `psa_ecc_curve_t` and `psa_dh_group_t` as deprecated aliases for the new type names.

File: include/psa/crypto_compat.h [hunk 4]
```diff
@@ -115,79 +115,110 @@ MBEDTLS_PSA_DEPRECATED static inline psa_status_t psa_asymmetric_verify( psa_key
...
-    MBEDTLS_DEPRECATED_CONSTANT( psa_dh_group_t, PSA_DH_GROUP_RFC7919 )
+    MBEDTLS_DEPRECATED_CONSTANT( psa_dh_family_t, PSA_DH_FAMILY_RFC7919 )
+
+/*
+ * Diffie-Hellman families that changed name due to PSA spec.
+ */
+#define PSA_DH_GROUP_RFC7919 \
+    MBEDTLS_DEPRECATED_CONSTANT( psa_dh_family_t, PSA_DH_FAMILY_RFC7919 )
+#define PSA_DH_GROUP_CUSTOM \
+    MBEDTLS_DEPRECATED_CONSTANT( psa_dh_family_t, PSA_DH_FAMILY_CUSTOM )
```
Comment (gilles-peskine-arm) [inline include/psa/crypto_compat.h:222]: Missing: old names for the renamed macros `PSA_KEY_TYPE_GET_CURVE` and `PSA_KEY_TYPE_GET_GROUP`.

File: ChangeLog.d/psa_dh_macros.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+API changes
```
Comment (gilles-peskine-arm) [inline ChangeLog.d/psa_dh_macros.txt:1]: All of the changes in this PR (ECC and DH) are very closely related and should be listed as a single bullet point.

> In the PSA API, rename the types of elliptic curve and Diffie-Hellman group families to psa_ecc_family_t and psa_dh_family_t, in line with the PSA Crypto API specification version 1.0.0. Rename associated macros as well.

Possibly followed by the list of macro renamings, but I didn't always bother earlier on.

File: ChangeLog.d/psa_ecc_macros.txt (hunk 1 above)
Comment (paul-elliott-arm) [reply ChangeLog.d/psa_ecc_macros.txt:3]: Done.

File: ChangeLog.d/psa_ecc_macros.txt (hunk 2 above)
Comment (paul-elliott-arm) [reply ChangeLog.d/psa_ecc_macros.txt:2]: Ok, fixed in comments as well.

Comment (paul-elliott-arm) [reply ChangeLog.d/psa_ecc_macros.txt:2]: Done.

File: include/psa/crypto_compat.h (hunk 3 above)
Comment (paul-elliott-arm) [reply include/psa/crypto_compat.h:54]: Done

File: include/psa/crypto_compat.h (hunk 4 above)
Comment (paul-elliott-arm) [reply include/psa/crypto_compat.h:222]: Done

File: ChangeLog.d/psa_dh_macros.txt (hunk 5 above)
Comment (paul-elliott-arm) [reply ChangeLog.d/psa_dh_macros.txt:1]: Merged both Changelist entries into one.

**Comment by paul-elliott-arm** [comment]
CI Fails are due to ABI Change (and MbedOS failures)

**Pull request merged**